### PR TITLE
POPJAM-2979: Button Loading spinner state

### PR DESCRIPTION
**Pull request by Tbruni85**
## JIRA
https://jira.superawesome.tv/browse/POPJAM-2979

## Description
Added new *loading* state to *FlatButton* class

## Screenshots
spinner mov

### CODE REVIEW

File: PopJamUI/Classes/GenericViews/FlatButton.swift [hunk 1]
```diff
@@ -146,4 +161,17 @@ extension UIButton {
...
         setBackgroundImage(colorImage, for: state)
     }
+    
+    func isLoading() {
+        
+        layer.borderColor = UIColor().PJGray().cgColor
+        layer.borderWidth = 1
+        setBackgroundColor(.clear, for: .disabled)
+        backgroundColor = .clear
+        let spinnigLoader = UIImage.gifImageWithName("loading_grey")
```
Comment (me2k) [inline PopJamUI/Classes/GenericViews/FlatButton.swift]: this could cause a problem outside the pod

**Comment by devgabrielcoman** [comment]
@Tbruni85 I get that. And not saying it makes things more complicated or we've changed **everything**. Just saying that it's deceptively simplifying our lives. 
Imagine tomorrow, for the NBCU app, Mike comes and says:
- We want the enabled button, once you've added the username & password, to be a bright rainbow colour instead of this blue colour. That makes sense, since enabled / disabled are traditional button state.s

Now, if Mike comes over and says:
- We don't want a spinner at all but we want a progress bar - then what?
Should the progress bar sit inside the button?
Or should it sit outside the button, a class of its own?

I imagine you'd agree it should sit on its own and deal with its own drawing, resizing, colours, etc.

The same for the spinner. It's easy to pluck a spinner in the button since it's a square gif and "somehow makes sense and makes our lives easier".
But change the spinner with any other loading mechanism and you'll see it doesn't make any more sense.

And if it doesn't make sense, then it follows the spinner should also be outside, since showing it is a function of the data being loaded & processed, not the user tapping the button.

File: PopJamUI/Classes/GenericViews/FlatButton.swift (hunk 1 above)
Comment (Tbruni85) [reply PopJamUI/Classes/GenericViews/FlatButton.swift]: the spinner is not meant to be customisable at the moment, so the gif will reside on the sdk side for now and it won't cause any issue.

File: PopJamUI/Classes/CreatePassword/Views/CreatePasswordView.swift [hunk 2]
```diff
@@ -231,7 +237,7 @@ public class CreatePasswordView: UIView, CreatePasswordViewModelDelegate {
     private func startPopjammingTapped(sender: UIButton) {
         loadingState(isLoading: true)
         // request the view model to create a new user
-        viewModel.createUser(password: passwordTextView.textField.text ?? "")
+        //viewModel.createUser(password: passwordTextView.textField.text ?? "")
```
Comment (me2k) [inline PopJamUI/Classes/CreatePassword/Views/CreatePasswordView.swift]: commented out code here... you know what to do

File: PopJamUI/Classes/GenericViews/FlatButton.swift [hunk 3]
```diff
@@ -132,6 +141,12 @@ public class FlatButton: UIButton {
             print("Wrong button type passed to init")
         }
     }
+    
+    // MARK: Actions
+    private func toggleLoadingState(isLoading: Bool) {
```
Comment (devgabrielcoman) [inline PopJamUI/Classes/GenericViews/FlatButton.swift]: I guess this isn't use anymore?

**Pull request merged**